### DISCUSSION TITLE
Windows Server 2003 x64 SP2 indefinitely looks for updates

### DISCUSSION THREAD

**Discussion by nhubbard** [LegacyUpdate/LegacyUpdate#262] [Questions & Help] [answered]
Hello! I'm running a Windows Server 2003 VM in a Proxmox instance for a security testing class, and I've been trying to use Legacy Update to patch the VM.

However, unlike the Server 2008 and Windows 7 VMs I have running on the same Proxmox instance, the Server 2003 system has been stuck on the "Checking for the latest updates for your computer..." screen for more than 24 hours now.

There's nothing seemingly happening on the machine, and all of my attempts to restart it, including stopping and reinstalling Legacy Update, have not changed the behavior. Is there any known issue with Server 2003? I've been trying to update this VM, but I don't want to switch to XP x32 if I don't have to...

**Comment by kirb** [comment, marked as answer]
That is pretty unusual. In my experience, Server 2003 can be finicky to start, including a few errors (usually “External cab file processing completed with some errors”), but the update check never finishing is strange. I would say 2003 should take no more than about 10 mins on a clean install with no additional Microsoft software installed yet.

What other Microsoft products do you have installed on this system? Office, Visual Studio, SQL Server, .NET, C++ runtimes, etc? Particularly Office 2007/2010 are problematic. There are some known workarounds to fix them breaking updates - try [this guide](https://msfn.org/board/topic/184136-fix-windows-update-stuck-on-checking-for-updates-on-vista-and-xp-well-kinda-after-installing-office-2010-sp2/) for 2010. You may also just find it less annoying to uninstall some stuff, see if it works, install updates if it does, and then install those apps again. It would at least get the machine partly up to date.

I’d recommend resetting the Windows Update database on the machine so it can start over. To do that:

1. Run in command prompt: `net stop wuauserv` (or stop Automatic Updates from the Services administrative tool)
2. Go to C:\Windows and delete the SoftwareDistribution folder.
3. Run Legacy Update again.

Hope you can figure this out, because your class sounds interesting!

**Reply by nhubbard** [reply]
I ended up switching it over to XP x32, which worked perfectly. The only reason I installed Server 2003 was the lack of drivers for XP x64. It was also acting really picky about the hardware selection on 2003, and I know that XP has an official guide for Proxmox.

**Reply by kirb** [reply]
Definitely you’ll have better compatibility on XP 32-bit/version 2002 compared to x64/version 2003. Haven’t tried 2003 with KVM/SPICE, but XP runs great with it in my experience. You may find VMware better for running OSes that aren’t well-supported by KVM/SPICE, but since you’re using Proxmox I imagine that might not be the best alternative (and ESXi doesn’t have a free version any more). But in general, 32-bit XP is the far more popular one, should be more compatible with some things, and might work a bit better for what you’re specifically doing.
